File: Stack/10828/10828.java
import java.util.Scanner;

class Node {
    public int data;
    public Node nextNode;

    public Node(int data) {
        this.data = data;
        this.nextNode = null;
    }

    public void linkNode(Node node) {
        this.nextNode = node;
    }

    public int getData() {
        return this.data;
    }

    public Node getNextNode() {
        return this.nextNode;
    }
}

class Stack {
    static int size;
    Node top;

    Stack() {
        this.top = null;
        this.size++;
    }

    public void push(int data) {
        Node node = new Node(data);
        node.linkNode(top);
        top = node;
    }

    public int pop() {
        if (isEmpty())
            throw new ArrayIndexOutOfBoundsException();
        else {
            int temp = top.data;
            top = top.getNextNode();
            this.size--;
            return temp;
        }
    }

    public boolean isEmpty() {
        return top == null;
    }

    public int top() {
        return top.getData();
    }
}

public class Main {
    

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        Stack st = new Stack();
        String order_str = "";
        int order_num = 0;

        int N = sc.nextInt();
        for(int i=0;i<N;i++) {
            order_str = sc.next();

            if(order_str.equals("push") == true) {
                order_num = sc.nextInt();
                st.push(order_num);
            }
            else if(order_str.equals("pop") == true) {
                System.out.println(st.pop());
            }
            else if(order_str.equals("size") == true) {
                System.out.println(st.size);
            }
            else if(order_str.equals("empty") == true) {
                if(st.isEmpty()) {
                    System.out.println(1);
                }
                else {
                    System.out.println(0);
                }
            }
            else {
                System.out.println(st.top());
            }
        }

        sc.close();
    }   
}

File: Stack/10828/10828.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct _Node {
	int data;
	struct _Node *next;
}Node;
int size = 0;

Node* push(Node* stack, int data);
int pop(Node** stack);
int isEmpty(Node* stack);
int peek(Node* stack);

int main() {
	Node* stack = NULL;
	int N = 0;
	int order_data = 0;

	scanf("%d", &N);
	for (int i = 1; i <= N; i++) {
		char* order_str = (char *)malloc(sizeof(char)*6);

		scanf("%s", order_str);
		if (strcmp(order_str, "push") == 0) {
			scanf("%d", &order_data);
			stack = push(stack, order_data);
		}
		else if (strcmp(order_str, "pop") == 0) {
			printf("%d\n", pop(&stack));
		}
		else if (strcmp(order_str, "size") == 0) {
			printf("%d\n", size);
		}
		else if (strcmp(order_str, "empty") == 0) {
			printf("%d\n", isEmpty(stack));
		}
		else {
			printf("%d\n", peek(stack));
		}

		free(order_str);
	}

	return 0;
}

Node* push(Node* stack, int data) {
	size++;
	Node *temp = (Node *)malloc(sizeof(Node));
	temp->data = data;
	temp->next = NULL;

	if (stack != NULL)
		temp->next = stack;

	return temp;
}
int pop(Node** stack) {
	if (*stack == NULL)
		return -1;
	else {
		size--;
		Node* temp_Node = *stack;
		*stack = (*stack)->next;

		int temp_data = temp_Node->data;
		free(temp_Node);

		return temp_data;
	}
}
int isEmpty(Node* stack) {
	if (stack == NULL)
		return 1;
	else
		return 0;
}
int peek(Node* stack) {
	if (stack == NULL)
		return -1;
	else
		return stack->data;
}